src/Compilers/Core/Portable/Text/StringBuilderText.cs

```diff
@@ -89,5 +91,45 @@ public override void CopyTo(int sourceIndex, char[] destination, int destination
         {
             _builder.CopyTo(sourceIndex, destination, destinationIndex, count);
         }
+
+        public override void Write(TextWriter textWriter, TextSpan span, CancellationToken cancellationToken = default)
+        {
+#if NETCOREAPP
+            if (span.Start == 0 && span.Length == this.Length)
+            {
+                textWriter.Write(_builder);
+            }
+            else
+            {
+                if (span.Length == 0)
+                    return;
+
+                int chunkOffset = 0;
+
+                foreach (var chunk in _builder.GetChunks())
+                {
+                    var startFromChunk = span.Start - chunkOffset;
```

Comment:
not sure how to read the term "startFromChunk".  sounds like a boolean from teh name.  can you clarify what this means?

Reply:
It's the starting index of the span, with respect to the current chunk (as opposed to the start of the text). How would you name this?

Reply:
`var startingIndexInCurrentChunk`?

Reply:
@CyrusNajmabadi even though it might not actually be inside that chunk?

Reply:
I mean you could say that span.Start sounds like a boolean or an action. The variable is defined right away and it's just one minus, if you don't get it at the first sight, you'd be able to see what it is from the initialization? It's kinda weird that the name would be almost twice as long as the expression it's supposed to shorten.

Reply:
i'm still not really sure what's going on here.  it doesn't help that this expression is built from another variable i'm unclear about "offset".

Reply:
@jaredpar Do you have any suggestions for a better name?